bluepyefe/ecode/sAHP.py

```diff
@@ -109,10 +104,98 @@ def interpret(self, t, current, config_data, reader_data):
         amp2_value = numpy.median(smooth_current[self.tmid : self.tmid2]) - self.hypamp
         self.set_amplitudes_ecode("amp2", config_data, reader_data, amp2_value)
 
-        # Converting back to ms
-        for name_timing in ["ton", "tmid", "tmid2", "toff"]:
-            self.index_to_ms(name_timing, t)
-        self.tend = len(t) * self.dt
+    def step_detection(self, current, config_data, reader_data):
+
+        # Set the threshold to detect the step
+        noise_level = numpy.std(numpy.concatenate((self.current[:50], self.current[-50:])))
+        step_threshold = numpy.clip(4.5 * noise_level, 1e-5, numpy.max(self.current))
```

Comment:
Why 4.5 specifically?

Reply:
It was the value that worked for all the data I had available. Above and below it misses some steps.